### PR TITLE
Fee is a KI64Pair

### PR DESCRIPTION
**Pull request by jaekwon**
* There's little difference between []byte and string (in most languages you can cast it) but there are small issues with string:
 -> https://github.com/protobuf-c/protobuf-c/issues/203
 -> We already use KVPair in several places and it's always been {[]byte, []byte} (e.g. we should refactor to eventually just use tmlibs/common/KVPair.  Using gogoproto we can do that now).
 -> Protobuf3 strings are supposed to be only utf8 or 7-bit ansii but for something called "KVPair", we don't want to be restricted to unicode or ansii characters.
 -> Perhaps using []byte makes us more vigilant in validating the input before accepting it, than be lulled into complacency w/ a string.

So this PR simplifies KVPair to just be ([]byte, []byte), and mayb we can do another PR to just use tmlibs/common.KVPair instead, or do it here.

It also introduces a more restrictive type, KI64Pair.  We can later introduce KI32Pair, KU64Pair, etc.

### CODE REVIEW

File: types/messages_test.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ func TestMarshalJSON(t *testing.T) {
 		Code: 1,
 		Data: []byte("hello"),
 		Gas:  43,
-		Fee:  12,
+		Fee:  KI64Pair{[]byte("pho"), 12},
```
Comment (odeke-em) [inline types/messages_test.go]: Hahah SF life ;)

**Comment by jaekwon** [comment]
It doesn't affect us but it will affect anyone writing C applications, if we ever put a null character in the KVPair "string".

**Comment by odeke-em** [comment]
Gotcha gotcha, great point.

**Comment by ebuchman** [comment]
Presumably the idea here is to put the coin denomination in as the key?

I think part of the motivation for the previous KVPair structure was to enable DeliverTx tags to have values that are either strings or integers - this change would force them to be strings (well, bytes, but same same). Is that what we want ? 

Otherwise this looks good, except that we may actually want int128 ...

**Comment by jaekwon** [comment]
The pubsub query could include casting of bytes into anything... so maybe for comparison you need to first cast it like "int64(value) < 123" or something. @melekes

**Comment by melekes** [comment]
yeah, we'll need to cast before making a comparison

https://github.com/tendermint/tmlibs/blob/master/pubsub/query/query.go#L228-L249

**Pull request merged**